viewing_party/party.py

```diff
@@ -0,0 +1,254 @@
+            for movie in friends_watched_list:
+                friends_movie_title = movie["title"]
+                # if a movie from user's unwatched movie list is in a friend's watched movies list
+                #  add the movie title and details (such as host or genre) to a list of user's unwatched movies
+                if unwatched_movie_title == friends_movie_title:
+                    movie_details = tuple(movie.values())
+                    unwatched_movies_and_details.add(movie_details)
+    
+    # return the list of movies that only the user's friends have watched including relevant details
+    return unwatched_movies_and_details
+
+def get_available_recs(user_data):
+    
+    # get a list of movies and their streaming services only the user's friends have watched
+    unwatched_movies_and_hosts = get_friends_unique_watched_with_details(user_data)
+    
+    available_recommended_movies = []
+
+    # loop over the user's list of unwatched movies and their streaming services 
+    for title, host in unwatched_movies_and_hosts:
+        # if the unwatched movie is hosted on a service that the user is subscribed to,
+        #  append the movie and service details to the list of recommended movies
+        if host in user_data["subscriptions"]:
+            available_recommended_movies.append({"title": title, "host": host})
+
+    # return the list of movies that are hosted by a service the user is subscribed to that they have not watched but their friends have
+    return available_recommended_movies
+
+def get_new_rec_by_genre(user_data):
+    
+    # get a list of movies and their genres that the user's friends have watched but the user has not
+    unwatched_movies_and_genres = get_friends_unique_watched_with_details(user_data)
+    
+    # get the genre the user watches the most
+    favorite_genre = get_most_watched_genre(user_data)
+
+    recommended_movies_in_favorite_genre = []
+
+    # loop over user's friend's list of unwatched movies and their genres 
+    for movie in unwatched_movies_and_genres:
+        genre = movie[1]
+        # if the unwatched movie is the same genre as the user's favorite genre,
+        #  append the movie and genre to the recommended movies list
+        if genre == favorite_genre:
+            recommended_movies_in_favorite_genre.append({"title": movie[0], "genre": genre})
+
+    # return the list of movies that only the user has watched that are in the user's favorite genre
+    return recommended_movies_in_favorite_genre
+
+def get_rec_from_favorites(user_data):
+
+    # get a list of movies only the user has watched
+    friends_unwatched_movies = get_unique_watched(user_data)
+    recommended_movies_from_favorites = []
+    users_favorites = user_data["favorites"]
+
+    # loop over each movie that the user's friends have not watched
+    for unwatched_movie in friends_unwatched_movies:
+        movie = unwatched_movie["title"]
```

Comment:
Consider `movie_title` here, since this is a reference to the title string, not a full movie record.